### DISCUSSION TITLE
Ironpython parallelism and library compatibility

### DISCUSSION THREAD

**Discussion by tal32123** [IronLanguages/ironpython3#1806] [Q&A]
Just a quick question, does ironpython support parallel processing. I am trying to use parallel foreach from C# to call python code which uses numpy/opencv/scikit/etc. I see that the GIL will prevent true multithreading/parallelism? Is this correct? Is it in the roadmap to get rid of the GIL as it seems to be an experimental python feature.

I saw a question from a few years ago that said ironpython won't work with opencv so also trying to ensure that my use case will work?

**Comment by slozier** [comment]
IronPython does not have a GIL, unfortunately it doesn't work with numpy/opencv/scikit/etc so your use case won't work.
